UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Matrix.cs

```diff
@@ -311,6 +311,27 @@ public IEnumerable<RegisterTile> GetRegisterTile(Vector3Int localPosition, bool
 		return (isServer ? ServerObjects : ClientObjects).Get(localPosition);
 	}
 
+	public IEnumerable<T> GetAs<T>(Vector3Int localPosition, bool isServer) where T : class
+	{
+		if (!(isServer ? ServerObjects : ClientObjects).HasObjects(localPosition))
+		{
+			return Enumerable.Empty<T>(); //?
+		}
+
+		var filtered = new List<T>();
+		foreach (RegisterTile t in (isServer ? ServerObjects : ClientObjects).Get(localPosition))
+		{
+			T x = t as T;
+			if (x != null)
```

Comment:
```suggestion
			if (t is T x)
```
Does that work and avoid the null?

Reply:
I don't know if it technically casts twice there though that's the thing?

Reply:
Dont think it would

Reply:
odd IDE isn't recommending the change anymore